include/nitro/broken_options/option/group.hpp

```diff
@@ -41,34 +44,190 @@ namespace broken_options
+                                       const std::string& description = std::string(""))
+        {
+            if (all_argument_names_.find(name) != all_argument_names_.end())
+            {
+                auto search = options_.find(name);
+                if (search == options_.end())
+                {
+                    raise<parser_error>(
+                        "Trying to redefine argument name in a different group. Name: ", name);
+                }
+                else
+                {
+                    return search->second;
+                }
+            }
+
+            all_argument_names_.emplace(name);
+            return options_
+                .emplace(std::piecewise_construct, std::forward_as_tuple(name),
+                         std::forward_as_tuple(name, description))
+                .first->second;
+        }
+
+        broken_options::multi_option& multi_option(const std::string& name,
+                                                   const std::string& description = std::string(""))
+        {
+            if (all_argument_names_.find(name) != all_argument_names_.end())
+            {
+                auto search = multi_options_.find(name);
+                if (search == multi_options_.end())
+                {
+                    raise<parser_error>(
+                        "Trying to redefine argument name in a different group. Name: ", name);
+                }
+                else
+                {
+                    return search->second;
+                }
+            }
+
+            all_argument_names_.emplace(name);
+            return multi_options_
+                .emplace(std::piecewise_construct, std::forward_as_tuple(name),
+                         std::forward_as_tuple(name, description))
+                .first->second;
+        }
+
+        broken_options::toggle& toggle(const std::string& name,
+                                       const std::string& description = std::string(""))
+        {
+            if (all_argument_names_.find(name) != all_argument_names_.end())
+            {
+                auto search = toggles_.find(name);
+                if (search == toggles_.end())
+                {
+                    raise<parser_error>(
+                        "Trying to redefine argument name in a different group. Name: ", name);
+                }
+                else
+                {
+                    return search->second;
+                }
+            }
+
+            all_argument_names_.emplace(name);
+            return toggles_
+                .emplace(std::piecewise_construct, std::forward_as_tuple(name),
+                         std::forward_as_tuple(name, description))
+                .first->second;
+        }
+
+        std::map<std::string, broken_options::option&> get_all_options()
+        {
+            std::map<std::string, broken_options::option&> tmp;
+            for (auto& it : options_)
```

Comment:
Replace this loop with `tmp.insert(iter, iter)`

Reply:
Impossible:
Cast from broken_options::option to broken_options::option&